server/src/main/java/org/elasticsearch/search/profile/query/CollectorResult.java

```diff
@@ -49,85 +50,27 @@ public class CollectorResult implements ToXContentObject, Writeable {
     private static final ParseField TIME_NANOS = new ParseField("time_in_nanos");
     private static final ParseField CHILDREN = new ParseField("children");
 
-    /**
-     * A more friendly representation of the Collector's class name
-     */
-    private final String collectorName;
-
-    /**
-     * A "hint" to help provide some context about this Collector
-     */
-    private final String reason;
-
-    /**
-     * The total elapsed time for this Collector
-     */
-    private final long time;
-
-    /**
-     * A list of children collectors "embedded" inside this collector
-     */
-    private final List<CollectorResult> children;
-
     public CollectorResult(String collectorName, String reason, long time, List<CollectorResult> children) {
-        this.collectorName = collectorName;
-        this.reason = reason;
-        this.time = time;
-        this.children = children;
+        super(collectorName, reason, time, new ArrayList<>(children));
```

Comment:
there's a couple of places where these typed list are a bit of a headache, I ended up creating another list to resolve silly compiler errors. Can you think of better ways?

Reply:
As this is for profiling, I think this is ok and it is safer. Generics can be a bit of a pain.